iothub/device/src/Pipeline/ExceptionRemappingHandler.cs

```diff
@@ -128,56 +130,50 @@ private static bool IsNetwork(Exception singleException)
+        private async Task<T> RunWithExceptionRemappingAsync<T>(Func<Task<T>> asyncOperation)
         {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"{nameof(ExceptionRemappingHandler)}.{nameof(ExecuteWithExceptionRemappingAsync)}");
+
             try
+            {
+                return await asyncOperation().ConfigureAwait(false);
+            }
+            catch (Exception ex) when (ex is not IotHubClientException && !Fx.IsFatal(ex))
             {
                 if (Logging.IsEnabled)
-                    Logging.Enter(this, $"{nameof(ErrorDelegatingHandler)}.{nameof(ExecuteWithErrorHandlingAsync)}");
+                    Logging.Error(this, $"Exception caught: {ex}");
 
-                try
+                if (IsSecurityExceptionChain(ex))
                 {
-                    return await asyncOperation().ConfigureAwait(false);
+                    Exception innerException = (ex is IotHubClientException) ? ex.InnerException : ex;
+                    throw new IotHubClientException("TLS authentication error.", IotHubClientErrorCode.TlsAuthenticationError, innerException);
                 }
-                catch (Exception ex) when (!Fx.IsFatal(ex))
+                else if (IsNetworkExceptionChain(ex))
+                {
+                    throw new IotHubClientException("A transient network error occurred; please retry.", IotHubClientErrorCode.NetworkErrors, ex);
+                }
+                else
                 {
                     if (Logging.IsEnabled)
-                        Logging.Error(this, $"Exception caught: {ex}");
-
-                    if (IsSecurityExceptionChain(ex))
-                    {
-                        Exception innerException = (ex is IotHubClientException) ? ex.InnerException : ex;
-                        throw new IotHubClientException("TLS authentication error.", IotHubClientErrorCode.TlsAuthenticationError, innerException);
-                    }
-                    // For historic reasons, part of the Error handling is done within the transport handlers.
-                    else if (ex is IotHubClientException hubEx
-                        && hubEx.ErrorCode is IotHubClientErrorCode.NetworkErrors)
-                    {
-                        throw;
-                    }
-                    else if (IsNetworkExceptionChain(ex))
-                    {
-                        throw new IotHubClientException("Transient network error occurred; please retry.", IotHubClientErrorCode.NetworkErrors, ex);
-                    }
-                    else
-                    {
-                        throw;
-                    }
+                        Logging.Error(this, $"Unmapped exception {ex.GetType()}");
+
+                    throw new IotHubClientException("An unexpected exception occurred. See the inner exception for more details.", IotHubClientErrorCode.Unknown, ex);
```

Comment:
I figured we should default to creating an IotHubClientException with the real one as the inner, rather than let it go through and log the error.

Thoughts?